platforms/mpi.jl

```diff
@@ -11,7 +11,7 @@ const augment = raw"""
     function augment_mpi!(platform)
         # Doesn't need to be `const` since we depend on MPIPreferences so we
         # invalidate the cache when it changes.
-        binary = get(preferences, "binary", Sys.iswindows() ? "MicrosoftMPI_jll" : "MPICH_jll")
+        binary = get(preferences, "binary", Sys.iswindows(platform) ? "MicrosoftMPI_jll" : "MPICH_jll")
```

Comment:
Can you add a brief comment. E.g. the logic will no longer be in-sync with MPIPreferences

https://github.com/JuliaParallel/MPI.jl/blob/4ae3df31c2aa521dab8603f974363e965646c795/lib/MPIPreferences/src/MPIPreferences.jl#L26